airflow/providers/common/sql/hooks/sql.py

```diff
@@ -111,6 +112,15 @@ def fetch_one_handler(cursor) -> list[tuple] | None:
         return None
 
 
+@contextmanager
+def suppress_and_warn(*exceptions: type[BaseException]):
+    """Context manager that suppresses the given exceptions and logs a warning message."""
+    try:
+        yield
+    except exceptions as e:
+        warnings.warn(f"Exception suppressed: {e}\n{traceback.format_exc()}", category=UserWarning)
+
+
```

Comment:
This one doesn't use in `common.sql`, so it should be moved into the specific provider, rather than keep it in `common.sql`

Reply:
I moved it to jdbc hook

Reply:
It could be nice thought to have the suppress_and_warn method in the common sql provider, as we could then also reuse it (after some refactoring allowing a custom warning message to be passed) in the following method and then we would be DRY (e.g. not repeating ourselves):

```
    def _make_common_data_structure(self, result: T | Sequence[T]) -> tuple | list[tuple]:
        """Ensure the data returned from an SQL command is a standard tuple or list[tuple].

        This method is intended to be overridden by subclasses of the `DbApiHook`. Its purpose is to
        transform the result of an SQL command (typically returned by cursor methods) into a common
        data structure (a tuple or list[tuple]) across all DBApiHook derived Hooks, as defined in the
        ADR-0002 of the sql provider.

        If this method is not overridden, the result data is returned as-is. If the output of the cursor
        is already a common data structure, this method should be ignored.
        """
        # Back-compatibility call for providers implementing old ´_make_serializable' method.
        with contextlib.suppress(AttributeError):
            result = self._make_serializable(result=result)  # type: ignore[attr-defined]
            warnings.warn(
                "The `_make_serializable` method is deprecated and support will be removed in a future "
                f"version of the common.sql provider. Please update the {self.__class__.__name__}'s provider "
                "to a version based on common.sql >= 1.9.1.",
                AirflowProviderDeprecationWarning,
                stacklevel=2,
            )

        if isinstance(result, Sequence):
            return cast(List[tuple], result)
        return cast(tuple, result)
```

It could become this then:

```
    def _make_common_data_structure(self, result: T | Sequence[T]) -> tuple | list[tuple]:
        """Ensure the data returned from an SQL command is a standard tuple or list[tuple].

        This method is intended to be overridden by subclasses of the `DbApiHook`. Its purpose is to
        transform the result of an SQL command (typically returned by cursor methods) into a common
        data structure (a tuple or list[tuple]) across all DBApiHook derived Hooks, as defined in the
        ADR-0002 of the sql provider.

        If this method is not overridden, the result data is returned as-is. If the output of the cursor
        is already a common data structure, this method should be ignored.
        """
        # Back-compatibility call for providers implementing old ´_make_serializable' method.
        with suppress_and_warn(
                AttributeError,
                message="The `_make_serializable` method is deprecated and support will be removed in a future "
                f"version of the common.sql provider. Please update the {self.__class__.__name__}'s provider "
                "to a version based on common.sql >= 1.9.1.",
                stacklevel=2,
            ):
            result = self._make_serializable(result=result)  # type: ignore[attr-defined]

        if isinstance(result, Sequence):
            return cast(List[tuple], result)
        return cast(tuple, result)
```

What do you think?

Reply:
Yes, that was the idea of my intial proposal - to move it there as it would be immediately useful for all those 24 providers. 

And yes - I agree with @Taragolis that this is more suitable for `common.utils` or whatever we might come up with - but well,  we do not have `common.utils` yet nor even the consensus of whether we need it and how to manage it. There was some vague idea about polyfills in the devlist discuussion (but I have no idea how to do it in the Python world and what it means for our build and release process). 

So we have no "perfect" solution - we have only "so-so" ones.

1) make it non-DRY and copy
2) reuse it via common.sql
3) (most complex) agree to, create and use `common.utils` 

I am good with all three approaches.  As long as someone will take on the 3) and prepares and takes lead of common.utils - I think it's the right time I stop leading and implementing this all myself.

I guess @Taragolis - pick your poison.

Reply:
(BTW. I am also super happy to help in implementing/reviewing 3) as long as somoene takes a lead there)

Reply:
I'm willing to start a new PR regarding option 3, as I have a feeling you guys prefer that approach, but I don't know if I will be able to lead it but can try.  I would have done it as option 2 but 3 is also fine for me, as long as at the end we come to a cleaner and DRY solution.

Reply:
>Ok. then what we should do if it is used in several providers - since we have no common.util - do you propose to copy it simply ? Less DRY, less coupling, more copy&paste. Or any other idea?

My proposal do not mix-in new feature changes in two different provider in the same time. 
- We do not know is it become useful into the other provider and would be
- We do not know side effect (in general, not in this case)

So if the purpose it to do into the single one/two provider, keep it in it do not need to put it into the `common.sql` it is not **common** and it is not **sql**

No required to replace one potential `common.utils` package by the other `common.sql`, no additional headache to release manager, because it required to bump version of dependency, and until last time it might be unknow which next version would be patch, minor and major

Reply:
We can definitely start with copying and the "common.utils" might come later - I am fine wit poison 1) and copying stuff. Definitely it's not worth introducing the common.util kind of approach for that single method,  buut we can later decide on common.utils and maybe a lot more common things. Or maybe we decide not to use it at all  (because yes - it introduces quite some hassle)

Reply:
There is two opinions fights to each other inside me:
First one told me: "hey! It would be awesome! It'll reduce codebase and might solve common mistake"
The second one whisper: "look at common sql, one simple change in common utility and it introduce bug in 50+ providers"

Reply:
The more I discover the code the more I think: "hey I saw this before, it would be nice to have that as a common util code".  Ofc there is the dependency issue, it's always a trade-off at some point, unless we could like include code dynamically like @potiuk said, but dunno if that's possible in Python.  In Java you could achieve a bit the same with an aspectj compiler at build phase.  Just wondering, if that new common util package would have no airflow dependencies, would it then still be an issue of dependency hell, as the origin of the dependency issues are due to the different airflow versions no?  But if the new package doens't depend on airflow (dunno if possible would have to try it), then new providers would only have an extra dependency on that package.  Or wy try a PR as draft and see what comes out of it dunno what you guys think, at some moment we'll have to think for a descent solution anyway.

Reply:
Yep. One day.